SU2_CFD/include/solvers/CNEMONSSolver.hpp

```diff
@@ -43,12 +43,22 @@
 class CNEMONSSolver final : public CNEMOEulerSolver {
 private:
 
-  su2double Prandtl_Lam,   /*!< \brief Laminar Prandtl number. */
-  Prandtl_Turb;            /*!< \brief Turbulent Prandtl number. */
+  su2double Prandtl_Lam,     /*!< \brief Laminar Prandtl number. */
+  Prandtl_Turb;              /*!< \brief Turbulent Prandtl number. */
  
-  su2double StrainMag_Max, Omega_Max; /*!< \brief Maximum Strain Rate magnitude and Omega. */
+  su2double StrainMag_Max,
+  Omega_Max;                 /*!< \brief Maximum Strain Rate magnitude and Omega. */
   su2double *primitives_aux; /*!< \brief Primitive auxiliary variables (Y_s, T, Tve, ...) in compressible flows. */
 
+  /*!
+   * \brief Compute the velocity^2, SoundSpeed, Pressure, Enthalpy, Viscosity.
+   * \param[in] solver_container - Container vector with all the solutions.
+   * \param[in] config - Definition of the particular problem.
+   * \param[in] Output - boolean to determine whether to print output.
+   * \return - The number of non-physical points.
+   */
+  unsigned long SetPrimitive_Variables(CSolver **solver_container,
+                                       CConfig *config, bool Output);
```

Comment:
It seems that this function is now repeated from NEMOEulerSolver to NEMONSSolver. check out what is done about this in the vanilla solvers, and follow that implementation (I think they use virtual for EulerSolver but best to check that carefully)

Reply:
You are right, the have a virtual in the CEulerSolver.hpp.  CNEMOEulerSolver.hpp has been updated appropriately.